ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+trying to keep the whole solution forward-compatible.
+
+With `SET_CODE_TX_TYPE=0x04` allowing EOA accounts to set contract code, EOA accounts will gain similar programmability
+as SCA (Smart Contact Account). A new standard, that provides EOA with highly-demanded AA features while potentially
+resolving some of the above challenges, will help bring in seamless user experience and drive mass adoptions.
+
+The above EIPs inspired the author to build an intent-centric standard for EOA smart accounts.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT
+RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### UserIntent bytes
+
+Each intent is a packed data structure containing sufficient information about operations the account owner wants to
+execute.
+
+| Field          | Type      | Description                                                                                                                                              |
+|----------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `sender`       | `address` | The wallet making the intent                                                                                                                             |
+| `standard`     | `address` | The `IStandard` implementation to validate and parse this `UserIntent`                                                                                   |
+| `header`       | `bytes`   | The metadata of this `UserIntent`, used by `standard`, using `bytes` type to keep flexibility                                                            |
+| `instructions` | `bytes[]` | The detailed content of this `UserIntent`, used by `standard` to determine the `Operation`s need to be executing, using `bytes` type to keep flexibility |
+| `signatures`   | `bytes[]` | Validatable signatures, used by `standard`                                                                                                               |
+
+But using `bytes[]` on functional interface is heavy especially when multiple intents are compound together. Therefore,
+the "UserIntent" object is designed to be a pack-encoded `bytes calldata userIntent` contains the above information
+
+| Section                         | Value Type | Description                                                     |
+|---------------------------------|------------|-----------------------------------------------------------------|
+| userIntent[0:20]                | `address`  | sender                                                          |
+| userIntent[20:40]               | `address`  | standard                                                        |
```

Comment:
You might want to expand "standard" here again.

Reply:
The "standard" here refers to the "standard" in the above table. I can add more descriptions if needed.